static/js-src/upcoming-page.js

```diff
@@ -0,0 +1,94 @@
+// Start fetching right away.
+const FEATURES_API_URL = '/api/v0/features';
+const CHANNELS_API_URL = '/api/v0/channels';
+const channelsArray = ['stable', 'beta', 'dev'];
+
+const channelsPromise = fetch(CHANNELS_API_URL)
+  .then((res) => res.text())
+  .then((res) => JSON.parse(res.substring(5)));
+
+document.querySelector('.show-blink-checkbox').addEventListener('change', e => {
+  e.stopPropagation();
+  document.querySelector('chromedash-schedule').showBlink = e.target.checked;
+});
+
+const header = document.querySelector('app-header-layout app-header');
+if (header) {
+  header.fixed = false;
+}
+
+async function init() {
+  // Prepare data for chromedash-schedule
+  const channels = await channelsPromise;
+  let featuresPromise = {};
+
+  channelsArray.forEach((channel) => {
+    let queryStringUrl = `${FEATURES_API_URL}?milestone=${channels[channel].version}`;
+    featuresPromise[channel] = fetch(queryStringUrl)
```

Comment:
Rather than defining the URLs and calling fetch() and stripping XSSI prefix in this file, it would be better to add API methods in static/js-src/cs-client.js.  If you follow the pattern of existing API calls in that file, your new functions will return a promise that you can use here.

Reply:
Okay. I have made the change in [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1426/commits/44628b1afb0f948d05237d20fca1eb912b1bb7ba) commit. 

Please review

Reply:
Looks good